DataMonitoring/pressure_analysis.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from time import sleep
+import os
+from scipy import signal
+from scipy.signal import savgol_filter
+from scipy.signal import lfilter
+
+#first test
+waterflowData = pd.read_csv(os.getcwd() + "/data/WaterFlowTest_08_08_16_42")
+tankData = waterflowData['low2'].to_numpy()
+amountOfData = len(tankData)
+currPressureData = np.asarray([0] * amountOfData)
+
+#second test
+waterflowData2 = pd.read_csv(os.getcwd() + "/data/waterflow_test_full_9_5_16_16")
+tankData2 = waterflowData2['low1'].to_numpy()
+amountOfData2 = len(tankData2)
+currPressureData2 = np.asarray([0] * amountOfData2)
+
+#third test
+waterflowData3 = pd.read_csv(os.getcwd() + "/data/waterflow_test_full_9_5_14_31 (1)")
+tankData3 = waterflowData3['low1'].to_numpy()
+amountOfData3 = len(tankData3)
+currPressureData3 = np.asarray([0] * amountOfData3)
+
+#function for detecting peaks in data
+##example: detect2(tankData)
+def detect2(data):
+    xaxis = np.arange(0,len(data),1)
+    print("Amount of Data Points:", len(data))
+    n = 5  # the larger n is, the smoother curve will be
+    b = [1.0 / n] * n
+    a = 1
+    yyy = lfilter(b,a,data)
+    dary = np.array([*map(float, data)])
+    dary -=np.average(dary)
+    step = np.hstack((np.ones(len(dary)), -1*np.ones(len(dary))))
+
+    dary_step = np.convolve(dary, step, mode='valid')
+    peaksall=[]
+    peaks = signal.find_peaks(dary_step, width=20)[0]
+    if len(peaks)>0:
+        peaksall.append(peaks)
+    print("Positive Peaks:", len(peaks))
+    peaks2 = signal.find_peaks(-dary_step, width=20)[0]
+    print("Negative Peaks:", len(peaks2))
+    if len(peaks2)>0:
+        peaksall.append(peaks2)
+    print("Total Peaks detected:", len(peaksall))
+    print(peaksall)
+    plt.figure()
+
+    plt.plot(dary)
+
+    plt.plot(dary_step/10)
+
```

Comment:
How about this function returns the location of the transitions, and you have a separate function to plot it. This modularity will make it very easy to combine with our other code later.